### PR TITLE
[12.0][MIG] mrp_bom_equivalent

### PR DESCRIPTION
**Pull request by AdriaGForgeFlow**
Standard migration of module mrp_bom_equivalent with some improvements.

Improvements:
- Product equivalent category is used instead of the general product category
- In the production stage, if the product in the BoM line is not available and the Use of equivalents is checked, the product with more priority in the same equivalent category that is available will be proposed.

Depends on:

- [x] [product-attribute/product_equivalent_category](https://github.com/OCA/product-attribute/pull/555)
- [x] [product-attribute/product_priority](https://github.com/OCA/product-attribute/pull/417)

cc ~ @ForgeFlow

### CODE REVIEW

**Comment by elvise** [comment]
Hi, how can I help for complete this task?

**Comment by JordiBForgeFlow** [comment]
@elvise funding helps!

**Comment by elvise** [comment]
> @elvise funding helps!

Ok, give me an idea about the effort

**Comment by elvise** [comment]
@AdriaGForgeFlow any good news for this PR ?

**Comment by AdriaGForgeFlow** [comment]
> @AdriaGForgeFlow any good news for this PR ?

Waiting for [product-attribute/product_equivalent_category](https://github.com/OCA/product-attribute/pull/555) to be merged

**Comment by elvise** [comment]
> > @AdriaGForgeFlow any good news for this PR ?
> 
> Waiting for [product-attribute/product_equivalent_category](https://github.com/OCA/product-attribute/pull/555) to be merged

@AdriaGForgeFlow seems merged :)

**Comment by kos94ok-3D** [comment]
@AdriaGForgeFlow Any news about this PR?

**Comment by kos94ok-3D** [comment]
@AdriaGForgeFlow Could you rebase the branch?

**Comment by AdriaGForgeFlow** [comment]
> @AdriaGForgeFlow Could you rebase the branch?

Done :+1:

File: mrp_bom_equivalent/models/product.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        )
+
+    @api.model
+    def search_read(
+        self, domain=None, fields=None, offset=0, limit=None, order=None
+    ):
+        if "nonequivalent_product_id" in self._context:
+            equivalent_category_id = self.browse(
+                self._context.get("nonequivalent_product_id")
+            ).equivalent_category_id
```
Comment (kos94ok-3D) [inline mrp_bom_equivalent/models/product.py:43]:
```suggestion
        if "nonequivalent_product_id" in self._context:
            equivalent_category_id = self.browse(
                self._context.get("nonequivalent_product_id")
            ).equivalent_categ_id
```

**Comment by kos94ok-3D** [comment]
@AdriaGForgeFlow This module does not work if manufacturing in 2 or 3 steps because we do not have components in pre-production location when we create MO

**Comment by AdriaGForgeFlow** [comment]
> @AdriaGForgeFlow This module does not work if manufacturing in 2 or 3 steps because we do not have components in pre-production location when we create MO

Hi @kos94ok-3D

Thanks for the review! I'm sorry but the intention of this Pull Request was to migrate the module to version 12. I can add your request to the ROADMAP but I don't have time now to look at it nor develop the changes. Nevertheless, you are more than welcome if you want to propose a PR with an improvement to the module.

**Pull request closed without merging**